File: snippets/python/src/closed_channel.py
import breez_sdk

def prepare_redeem_onchain_funds(sdk_services, fee_rate):
   
    try:
         # ANCHOR: prepare-redeem-onchain-funds
        destination_address = "bc1.."
        sat_per_vbyte = fee_rate
        req = breez_sdk.PrepareRedeemOnchainFundsRequest(destination_address, sat_per_vbyte)
        result =  sdk_services.prepare_redeem_onchain_funds(req)
        # ANCHOR_END: prepare-redeem-onchain-funds
        return result
    except Exception as error:
        print(error)
        raise

def redeem_onchain_funds(sdk_services, to_address, fee_rate):
    try:
        # ANCHOR: redeem-onchain-funds
        req = breez_sdk.RedeemOnchainFundsRequest(to_address, fee_rate)
        result = sdk_services.redeem_onchain_funds(req)
        # ANCHOR_END: redeem-onchain-funds
        return result
    except Exception as error:
        print(error)
        raise

File: snippets/go/closed_channel.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-go/breez_sdk"
)

func PrepareRedeemOnchainFunds(feeRate uint32) error {
	// ANCHOR: prepare-redeem-onchain-funds
	satPerVbyte := feeRate
	destinationAddress := "bc1.."
	req := breez_sdk.PrepareRedeemOnchainFundsRequest{SatPerVbyte: satPerVbyte, ToAddress: destinationAddress}
	prepareRedeemOnchainFundsResponse, err := sdk.PrepareRedeemOnchainFunds(req)
	if err != nil {
		return err
	}
	log.Printf("PrepareRedeemOnchainFundsRequest is %#v", prepareRedeemOnchainFundsResponse)
	// ANCHOR_END: prepare-redeem-onchain-funds
	return nil
}

func RedeemOnchainFunds(satPerVbyte uint32, toAddress string) error {
	// ANCHOR: redeem-onchain-funds
	req := breez_sdk.RedeemOnchainFundsRequest{SatPerVbyte: satPerVbyte, ToAddress: toAddress}
	redeemOnchainFundsResponse, err := sdk.RedeemOnchainFunds(req)
	if err != nil {
		return err
	}
	log.Printf("RedeemOnchainFunds error %#v", redeemOnchainFundsResponse)
	// ANCHOR_END: redeem-onchain-funds
	return nil
}
